lib/runtime/types.go

```diff
@@ -37,6 +37,26 @@ type NodeStorage struct {
 	PersistentStorage BasicStorage
 }
 
+// SetLocal persists a key and value into LOCAL node storage
+func (n *NodeStorage) SetLocal(k, v []byte) error {
+	return n.LocalStorage.Put(k, v)
+}
+
+// GetLocal retrieve a key and value from LOCAL node storage
+func (n *NodeStorage) GetLocal(k []byte) ([]byte, error) {
+	return n.LocalStorage.Get(k)
+}
+
+// SetPersistent persists a key and value into PERSISTENT node storage
+func (n *NodeStorage) SetPersistent(k, v []byte) error {
+	return n.PersistentStorage.Put(k, v)
+}
+
+// GetPersistent retrieve a key and value from PERSISTENT node storage
+func (n *NodeStorage) GetPersistent(k []byte) ([]byte, error) {
+	return n.PersistentStorage.Get(k)
+}
+
```

Comment:
it might be nice to refactor the offchain storage into its own API/package that both the runtime and the RPC can access, this doesn't need to be done here but maybe in the future. there's an issue for it: #1508

Reply:
I've outlined this in the issue #1776